substrafl/model_loading.py

```diff
@@ -147,93 +144,191 @@ def _get_train_task_from_round(client: substra.Client, compute_plan_key: str, ro
+        - the file specified in metadata.model_file
 
-def download_algo_files(
+    Args:
+        input_folder (os.PathLike): Path to folder containing the required files.
+        remote (bool): Wether the the instance to load is to load using a local method or a remote method.
+
+    Raises:
+        exceptions.LoadMetadataError: The metadata file must contains the model_file key
+        exceptions.LoadFileNotFoundError: At least one of the required file to load the instance is not found
+        exceptions.LoadLocalDependencyError: One of the dependency used by the instance is not installed within the
+          used environment
+
+    Returns:
+        Any: The serialized instance within the input_folder
+    """
+
+    folder = Path(input_folder)
+
+    metadata = _validate_folder_content(folder=folder)
+
+    _check_environment_compatibility(metadata=metadata)
+
+    try:
+        instance = _load_instance(gz_path=folder / metadata[FUNCTION_DICT_KEY], extraction_folder=folder, remote=remote)
+
+        if not remote:
+            loaded_instance = instance.load_local_state(folder / metadata[MODEL_DICT_KEY])
+        else:
+            loaded_instance = instance.load_model(folder / metadata[MODEL_DICT_KEY])
+
+    except ModuleNotFoundError as e:
+        raise exceptions.LoadLocalDependencyError(
+            "The instance from the given input folder requires the installation of "
+            "additional dependencies. Those can be found in "
+            f"{str(folder / 'substrafl_internal' / 'installable_library')}"
+            f"\nFull trace of the error: {e}"
+        )
+
+    return loaded_instance
+
+
+def _download_task_output_files(
     *,
     client: substra.Client,
     compute_plan_key: str,
     dest_folder: os.PathLike,
+    task_type: str,
+    identifier: OutputIdentifiers,
     round_idx: Optional[int] = None,
-):
-    """Download all the files needed to load the model:
+    rank_idx: Optional[int] = None,
+) -> None:
+    """Download all the files needed to load the SubstraFL instance:
 
         - hosted on the client organization
         - being part of the given compute plan
-        - being the result of the associated strategy after `round_idx` steps
+        - being the result of the associated strategy after `round_idx` steps or at rank `rank_idx`
 
     into memory.
 
     Those files are:
 
         - the function used for this task
-        - the output local state of the task
+        - the output state of the task
         - a metadata.json
 
-    Important:
-        This function supports only strategies with one train task for a given organization and round.
-
     Args:
         client (substra.Client): Substra client where to fetch the model from.
         compute_plan_key (str): Compute plan key to fetch the model from.
         dest_folder (os.PathLike): Folder where to download the files.
+        task_type (str): Type of the task to fetch. Allowed value are "init", "train" and "aggregate".
```

Comment:
minor but maybe we want that to be an `Enum`? 🤔

Reply:
100% agree, thanks. I created a schemas.py at the root of substrafl to enum the `TaskType`. It seems to me the best place to put it, but not 100% sure

Reply:
It seems reasonable to me too